cfme/utils/template/base.py

```diff
@@ -50,41 +64,28 @@ class ProviderTemplateUpload(object):
     Class variables:
         :var provider_type: type of initiated provider -- to be removed
         :var log_name: string to be displayed in logs.
-        :var image_patters: regex to be matched when stream URL is used
+        :var image_pattern: regex to be matched when stream URL is used
     """
     provider_type = None
     log_name = None
     image_pattern = None
 
-    def __init__(self, stream=None, provider=None, template_name=None,
-                 cmd_line_args=None, **kwargs):
+    def __init__(self, provider_key, stream, template_name, image_url=None, **kwargs):
         """
         Required parameters:
             :param stream: name of stream
-            :param provider: name of provider
+            :param provider_key: key of provider in yaml
             :param template_name: name of template to use
-
-        Optional parameters:
-            :param stream_url: custom URL to image directory of a stream
-            :param image_url: custom URL to exact image file
-            :param provider_data: custom AttrDict with provider data
+            :param image_url: custom URL to image directory of a stream, defaults to stream/stable
 
         Default values for custom parameters:
-            :param stream_url: cfme_data.basic_info.cfme_images_url[stream]
-            :param image_url: parsed with regex from stream_url web page
-            :param provider_data: cfme_data.management_systems[provider]
+            :param image_url: cfme_data.basic_info.cfme_images_url[stream]
         """
-        self._stream_url = kwargs.get('stream_url')
-        self._image_url = kwargs.get('image_url')
-        self._provider_data = kwargs.get('provider_data')
-
-        self._cmd_line_args = cmd_line_args
-
         self.stream = stream
-        self.provider = provider
+        self.provider_key = provider_key
```

Comment:
OPTIONAL I would try attrib in these scripts to make them simpler

Reply:
going to do all other comments first and then come back to this.